src/operations/connect.ts

```diff
@@ -51,8 +51,9 @@ export function connect(
   if (typeof options.srvHost === 'string') {
     return resolveSRVRecord(options, (err, hosts) => {
       if (err || !hosts) return callback(err);
-      for (const [index, host] of hosts.entries()) {
-        options.hosts[index] = host;
+
+      for (let index = 0; index < hosts.length; index++) {
```

Comment:
was there something wrong with the original for..of loop? if not, let's put it back exactly how it was so that it doesn't show in the changed files list